File: project files/server/src/models/DeliverySlot.js
const mongoose = require('mongoose');

const deliverySlotSchema = new mongoose.Schema(
  {
    date: { type: Date, required: true, index: true },
    timeWindow: { type: String, required: true },
    capacity: { type: Number, required: true, min: 0 },
    booked: { type: Number, default: 0, min: 0 },
    isActive: { type: Boolean, default: true }
  },
  { timestamps: true }
);

module.exports = mongoose.model('DeliverySlot', deliverySlotSchema);

File: spring_back/src/main/java/com/groceryapp/model/DeliverySlot.java
package com.groceryapp.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "delivery_slots")
public class DeliverySlot {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    @Column(name = "slot_date")
    private Instant date;
    private String timeWindow;
    private Integer capacity;
    private Integer booked;
    private Boolean isActive;
    private Instant createdAt;
    private Instant updatedAt;
}
